src/intrinsics/ecma262/JSON.js

```diff
@@ -328,16 +329,20 @@ function InternalGetTemplate(realm: Realm, val: AbstractObjectValue): ObjectValu
 function InternalJSONClone(realm: Realm, val: Value): Value {
   if (val instanceof AbstractValue) {
     if (val instanceof AbstractObjectValue) {
+      let callee = "global.JSON.parse(global.JSON.stringify(OBJECT))";
+      let [hash, args] = hashCall(callee, val);
       return realm.createAbstract(
```

Comment:
It's probably okay in this pull request, as InternalJSONClone is only used to construct the template object. But in general, one has to be careful: Calling `JSON.parse()` is always going to create a new object identity, and two such calls shouldn't be considered equivalent.